### PR TITLE
Speed up vote view

### PR DESCRIPTION
**Pull request by dpoirier**
Remove unnecessary queries, shorten code and make
it more idiomatic.

### CODE REVIEW

File: opendebates/views.py [hunk 1]
```diff
@@ -157,21 +164,13 @@ def vote(request, id):
...
-    except (IndexError, AttributeError):  # if related1 is None -> related1.id
-        related2 = None
-
     if request.method == "GET":
+        two_other_approved_ideas = list(Submission.objects.filter(
+            category=idea.category,
+            duplicate_of=None,
+            approved=True).exclude(id=idea.id)[:2]) + [None, None]
+        related1 = two_other_approved_ideas[0]
+        related2 = two_other_approved_ideas[1]
```
Comment (ironfroggy) [inline opendebates/views.py]: This is going to fetch and materialize many more submissions than it actually uses, can we avoid that? You could achieve similar with `itertools.chain(Submission..., [None, None])` and two calls of `next()`

Comment (dpoirier) [inline opendebates/views.py]: Won't the `[:2]` on the queryset limit to the first 2 results being returned?

Comment (ironfroggy) [inline opendebates/views.py]: I may have misread the placement of the `[:2]` and the closing `)`

**Comment by vkurup** [comment]
:+1:

**Pull request merged**